proposals/NNNN-android-oreo-foreground-services.md

```diff
@@ -0,0 +1,145 @@
+
+### 2. Router service with app in foreground for unknown BT devices
+
+#### Conditions
+
+- An SDL enabled app is in foreground on the phone
+- The app never established to register to the connected Bluetooth device
+- On the phone there is no router service provided by another app 
+
+#### Requirement
+
+1. The app should start the SDL related services as a regular service (not in foreground).
+
+2. The app should keep the services up and running for the time the app is in foreground.
+
+3. The app should stop the services when entering background.
+
+#### Comments
+
+This is required. Otherwise the app will never be prepared for SDL head units.
+
+### 3. App made it to connect to unknown BT device
+
+#### Conditions
+
+- An SDL enabled app is in foreground on the phone
+- The app never established to register to the connected Bluetooth device
+- On the phone there is no router service provided by another app 
+- App got connected and registered to the head unit
+
+#### Requirements
+
+1. The app should store that information
+2. The app should make the router service a foreground service
+
+#### Comments
+
+This is the end of the "first time" guide storing the information of the user owning an SDL capable device
+
+### 4. Router service with app in background for known SDL BT devices
+
+#### Conditions
+
+- An SDL enabled app is not in foreground on the phone
+- The app established to register to the connected Bluetooth device
+- On the phone there is no router service provided by another app 
+
+#### Requirements
+
+1. The app should start the router service as a foreground service including a notification.
+
+2. The router service should stay in foreground while being connected to that BT device (no timeout after 10 - 20 seconds)
+
+#### Comments
+
+As the app knows that the device supports SDL it can start the router service and keep is active/foregrounded
+
+### 5. App connect to router service when in background
+
+#### Conditions
+
+- An SDL enabled app is in foreground on the phone
+- The app never established to register to the connected Bluetooth device
+- On the phone another app has created the router service
+
+#### Requirements
+
+1. The app's proxy should connect to the router service
+ 
+2. The router service should tell the app if the connected BT device is known to support SDL
+
+#### Comments
+
+This behavior can exist if another SDL enabled app is in foreground. It does not neceesarily know if the BT device supports SDL.
+
+## Potential downsides
+
+The downside is the reduce experience for the first time use as the phone must be unlocked and the app must be started. It's expected that users might do this anyway if the know about the SDL support.
+
+## Impact on existing code
+
+Changes to the library are not identified but it's expected that this change will be possible in a minor version change. 

```

Comment:
These changes should be identified in the proposal, as part of the `Proposed solution` section.